src/wallet/wallet.cpp

```diff
@@ -6750,6 +6757,117 @@ bool CWallet::InitLoadWallet(const CChainParams& params, bool clearWitnessCaches
+
+            uint32_t lineNumber = 0;
+            uint32_t lastUsedLine;
+            while (recoverFile.good()) {
+                // Constrain parser to follow the multi-line format of z_exportwallet:
+                std::getline(recoverFile, line);
+                lineNumber++;
+
+                if (!line.empty()) {
+                    auto recoverPos = line.find("recovery_phrase=");
+                    if (recoverPos != std::string::npos) {
+                        // Check that file contains exactly one recovery phrase line
+                        assert(!recoveryPhrase.has_value());
+                        lastUsedLine = lineNumber;
+                        SecureString tmp = line.substr(recoverPos + 16, std::string::npos);
+                        boost::trim(tmp);
+                        recoveryPhrase = boost::algorithm::erase_all_copy(tmp, "\"");
+                    }
+
+                    // If a "language=" line appears, it must immediately follow
+                    // "recovery_phrase="
+                    auto languagePos = line.find("language=");
+                    if (languagePos != std::string::npos) {
+                        assert(recoveryPhrase.has_value());
+                        assert(lineNumber == (lastUsedLine + 1));
+                        lastUsedLine++;
+                        auto languageName = line.substr(languagePos + 9, std::string::npos);
+                        boost::trim(languageName);
+                        language = MnemonicSeed::LanguageFromName(languageName.c_str());
+                        if (!language.has_value()) {
+                            return UIError(strprintf(_("The given recovery phrase language '%s' is unknown."), languageName));
+                        }
+                    }
+
+                    // Look for an optional "fingerprint=" line that is
+                    // immediately following either "recovery_phrase=" or
+                    // "language=".
+                    auto fingerprintPos = line.find("fingerprint=");
+                    if (fingerprintPos != std::string::npos) {
+                        assert(recoveryPhrase.has_value());
+                        assert(lineNumber == (lastUsedLine + 1));
+                        lastUsedLine++;
+                        auto seedFpStr = line.substr(fingerprintPos + 12, std::string::npos);
+                        boost::trim(seedFpStr);
+                        uint256 seedFp;
+                        seedFp.SetHex(seedFpStr.c_str());
+                        fingerprint = seedFp;
+                    }
+
+                    // Syntax:
+                    //     num_accounts=INT
+                    //
+                    // Constrained to immediately follow fingerprint= (if present).
+                    //
+                    // If this line is not present, treat it as num_accounts=1.
+                    // If num_accounts=0, treat it as such (no automatic account recovery).
```

Comment:
I'm concerned that this will lead to loss of funds if someone recovers a wallet that had more than one account (and the dump file was written before v5.3.0 so it doesn't have a `num_accounts` field), and they don't immediately notice that the funds in accounts after the first are missing.

Reply:
Should we instead attempt to recover more than one account, and then remove it if we don't find any funds sent to any of the subsequent accounts? That gets a little tricky because we don't know how many accounts we should scan with, and it multiplies the cost of rescanning.

This isn't so much a loss-of-funds bug as it is a "perceived-loss-of-funds" issue; if someone encounters this then they can always attempt to reimport with a corrected value for `num_accounts` - it shouldn't be the case that someone deletes or otherwise makes their seed phrase unrecoverable just because they've performed a single recovery operation.